packages/vite/src/server.ts

```diff
@@ -42,6 +42,7 @@ export async function buildServer (ctx: ViteBuildContext) {
       noExternal: [
         ...ctx.nuxt.options.build.transpile,
         // TODO: Use externality for production (rollup) build
+        /\/esm\//,
```

Comment:
This check seems too broad. 
```suggestion
        /\/esm\/.*\.js$/,
```